### DISCUSSION TITLE
[forward compat?] Support Python 3.10

### DISCUSSION THREAD

**Discussion by webknjaz** [abhinavsingh/proxy.py#804] [Q&A]
**Describe the bug**

I've recently integrated proxy.py into the test suite of aiohttp. Now that we're trying to add Python 3.10 to the CI, the tests that interact with an instance of proxy.py get a TLS handshake timeout.

I haven't looked into it but my guess is that proxy.py may be incompatible with Python 3.10.

**To Reproduce**

1. Clone aio-libs/aiohttp
2. Run `make vvtest`

**Expected behavior**

It should work.

**Version information**
 - OS: Ubuntu with Python 3.10 activated
 - Browser: N/A
 - Device: GHA CI/CD
 - proxy.py Version 2.3.1

**Additional context**

https://github.com/aio-libs/aiohttp/pull/5927

**Screenshots**

N/A

**Comment by abhinavsingh** [comment]
Thank you @webknjaz for trying it on `Python 3.10`.  I haven't yet got my hands on `3.10`.  I'll try to jump on it in coming week and see how it goes.

**Comment by abhinavsingh** [comment]
@webknjaz I updated to `3.10` today.  All looks good locally, tests are passing and I am able to post this comment (all via `proxy.py` running on `3.10`).  I have also added official support for `3.10` in `proxy.py` CI.

I looked into the linked issue and IIUC tests at your end are seeing `ConnectionAbortedError: SSL handshake is taking longer than 60.0 seconds: aborting the connection` error.  I am unsure if this is linked to `proxy.py`, but we'll need to investigate deeply to confirm anything.

I'll give `make vvtest` a try and see if I can find anything useful.

**Comment by webknjaz** [comment]
Thanks, that should be helpful! I haven't had time to look into that specific error myself but if I'll be able to check it out, I'll report back here too.

**Comment by abhinavsingh** [comment]
@webknjaz Should we look back into it within `aiohttp` or do we need to get into `proxy.py` deeper.  In general, `proxy.py` seems to work fine.  I am now using it for more than a month on `Python 3.10`.  Wdyt?

**Comment by webknjaz** [comment]
Well, I'll only be able to check it in that CI once you make a release.

**Comment by abhinavsingh** [comment]
@webknjaz Wondering if it's ok to resolve this one now.

Probably will mark as resolve next time I come around this.  We can always reopen iff necessary :)

Once again thank you for all your help/contributions/support as we went around this.

I hope all is well at your end.  Best.

**Reply by webknjaz** [reply]
It seems like this is still happening with v2.4.3. And I'm seeing this problem under older CPython versions on Windows too: https://github.com/aio-libs/aiohttp/runs/8199178923?check_suite_focus=true#step:10:4910.

**Reply by abhinavsingh** [reply]
Thanks for cross verifying.  From first look, seems like issue with sharing fd between processes.  I'll give it a try once locally on a Windows system and see if I can reproduce it.  Will update back.

**Reply by webknjaz** [reply]
Thanks! It also appears to look like https://github.com/abhinavsingh/proxy.py/issues/1000.

**Reply by webknjaz** [reply]
@abhinavsingh have you had a chance to take a look? We have the tests marked as expected to fail currently but it'd be nice to turn that expectation to success.

**Reply by abhinavsingh** [reply]
Hi @webknjaz , Its been a busy week.  I'll try to pull up sometime over the coming weekend.  Best.
